umbraco/Umbraco-CMS
# Item tracking fixes

## Details
This PR fixes some of the issues mentioned as comments in https://github.com/umbraco/Umbraco-CMS/pull/11919, together with a bug where duplicate items were displayed on the items tracking result:
- Adding xdoc comments on public interfaces;
- Fixing some wording and other minor fixes;
- Removing duplicate items when displaying references. 

## Test
- Reference the same items in several different places;
- Make sure all the references appear;
- Make sure that when this item is part of bulk deletion/unpublishing or when deleting/unpublishing the parent of the item, it doesn't appear several times in the result since it has more than 1 reference.
